src/cli/download_handler.rs

```diff
@@ -84,15 +97,65 @@ impl DownloadHandler {
     }
 
     pub fn run(&self) -> HandlerResult {
+        let destination = self.selection_destination_for_install()?;
+        self.destination_may_not_dir(&destination)?;
```

Comment:
This should be done inside `maybe_install` function like it was before. This is a specific behaviour of the install feature.